src/api/services/memberInfo/MemberInfoService.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  GithubInfo,
+  StudyData,
+  Member,
+  Cohort,
+  Submission,
+  MemberInfo,
+} from "./types";
+import { IGithubApiClient } from "../github/interfaces";
+import { IMemberInfoService } from "./interfaces";
+import { handleError } from "../../common";
+import { GithubTree } from "../github/types";
+
+export class MemberInfoService implements IMemberInfoService {
+  constructor(
+    private readonly githubApiClient: IGithubApiClient,
+    private readonly config: GithubInfo,
+  ) {}
+
+  async getMemberInfo(): Promise<StudyData> {
+    try {
+      const members = await this.getAllTeamMembers();
+      const submissions = await this.getSubmissionData();
+      const memberInfoMap = this.createMemberInfoMap(members, submissions);
+
+      return {
+        total: members.length,
+        data: Object.values(memberInfoMap),
+      };
+    } catch (error) {
+      handleError("Error getting member info", error);
+      return { total: 0, data: [] };
+    }
+  }
+
+  private async getAllTeamMembers(): Promise<Member[]> {
+    const teams = await this.getTeams();
```

Comment:
`getTeamMembers()` 함수의 매개 변수의 이름도 `teamName`이네요.

```suggestion 
     const teamNames = await this.getTeamNames();
```